src/interfaces/swig/shogun.i

```diff
@@ -209,7 +209,7 @@ namespace shogun
 			// vector with correct inner type
 			if ($self->has(tag_vec_X))
 			{
-				SGVector<X> vec(mat.data(), mat.size(), false);
```

Comment:
Reference counting was turned off. What's weird is this didn't segfault, but consistently had the exact same random effects on each run.

Reply:
good catch!
No idea why I put that in .... ?